Update Button card documentation (#13807)
home-assistant/home-assistant.github.io: source/_lovelace/button.markdown

The Button card allows you to add buttons to perform tasks.

<p class='img'>

> <img src='/images/lovelace/lovelace_entity_button_card.png' alt='Screenshot of the button cards'>

Reviewer: Make the alt here the same as the description
```suggestion
<img src='/images/lovelace/lovelace_entity_button_card.png' alt='Screenshot of three Button Cards'>
```

Screenshot of three Button Cards.
</p>

> ## Card Settings

Reviewer: ```suggestion
### Card Settings
```

{% configuration_basic %}
Entity:
  description: The entity the card interacts with, for example `light.living_room`.
Name:

>   description: The button name. The default is the name of the entity, if an entity is selected.

Reviewer: ```suggestion
  description: The name that will appear on the button. If not set, the name will be the name of the entity, if an entity is selected.
```

Lets try to keep the wording similar for all cards.

Author: I’m using the [Google style guide](https://developers.google.com/style/tense) to inform best practice here. If we have to change it, my preference would be “The name that appears on the button”,  or to align with the next sentence, “The name shown on the card” - but that seems very wordy for something that only needs a short description. Since we are updating all the card documentation (or at least I thought we were??), this might be something to change across them all.

Reviewer: The setting is called Name... The button Name gives us no more detail about that settings. Something about it appearing on the button/card is good. I know its basically common sense but a name could be above the card or below etc. And otherwise we aren't giving any more details. Style it the way you want. My suggestions are more of a push to change it no exactly what it should be or what is best

Author: Hmmm... how about “The button name, which is displayed on the card below the icon. If this field is left blank and the card interacts with an entity, the button name defaults to the entity ID. If this field is left blank and the card does not interact with an entity, no name is displayed on the card.”

Reviewer: Yea except it defaults to the name of the entity not the entity ID

Icon:

>   description: The icon shown on the card. The default is the entity domain icon, if an entity is selected.

Reviewer: ```suggestion
  description: The icon shown on the card. If not set, the icon will be the entity domain icon, if an entity is selected.
```

> Show Name?:

Reviewer: ```suggestion
Show Name:
```

  description: A toggle to show or hide the button name.

> Show Icon?:

Reviewer: ```suggestion
Show Icon:
```

  description: A toggle to show or hide the icon.
Icon Height:

>   description: The height of the icon, in pixels.

Reviewer: ```suggestion
  description: The height of the icon, in pixels. (Ex. 50px)
```

Author: If I’m adding a new card, or editing an existing one, the editor already shows me that whatever I enter will be in pixels, so even that is almost redundant information. As soon as I enter a number, the icon height changes in the card preview. I’m not sure we’re adding much useful information by giving an example.

Reviewer: Good point 👍

Theme:

>   description: The card theme, which may be set to any installed theme.

Reviewer: ```suggestion
  description: Theme your card using any installed theme.
```

Can you also link `theme` to the them documentation

Author: Do we have a style guide other than what’s on the developer documentation standards page? Both the Microsoft style guide and the Google developer documentation style guide suggest writing parameters in the way we have here (starting with ‘the’ or ‘a/an’, with a description that follows). My preference would be to have them all written in the same way.

Reviewer: @frenck Is our style guide 😄 but starting with the sounds like a good guide to follow

Reviewer 2: We don't have a style guide on this yet. Both the Microsoft & Google guides are amazing, but we don't enforce it.

Tap Action:
  description: The action taken on card tap. For more information, see the [action documentation](/lovelace/actions/#tap-action).
Hold Action:

>   description: The action taken on card tap and hold. For more information, see the [action documentation](/lovelace/actions/#hold-action).

Reviewer: Card Tap? is this supposed to be just hold?

Author: I’m referencing the Actions page here, which describes it as a tap and hold. 
![F330832A-85B3-4272-8B54-12B23B95F95F](https://user-images.githubusercontent.com/64062738/88225289-55967e00-ccbe-11ea-9f2d-d712c41d4cf5.jpeg)

Reviewer: Gotcha. hmm maybe that is wrong then

@bramkragten is this correct wording? It only makes sense to be only hold right?

Reviewer 3: Tapped is weird anyway, what is a tap? A click is not a tap right? Yet it also works with a click.

But uuuuh... to hold it, you first have the tap/click it?
It is either long press or tap/touch/click/press & hold?

Author: We could move to click instead of tap I guess, but to be honest I think the meaning is pretty clear and a lot of people will have this UI set up on a touch device such as a tablet.  It also follows the naming convention in the YAML, which is `tap_action` and `hold_action`. But I do agree with @bramkragten that in order to hold, you must first tap.

Reviewer: I don't think anyone will be trying to hold the button without tapping/clicking it first :)

Reviewer 3: It's a generic name, google it 🙂

Author: I’ve left it as is for now. If we decide to change the Actions documentation, we can change the cards documentation to match?

{% endconfiguration_basic %}

## Code Editor or YAML mode

> 

Reviewer: ```suggestion

This is for if you use YAML mode or just prefer to use the YAML Code Editor in the UI
```

Link to the YAML mode docs

The Button card allows you to add buttons to perform tasks.

<p class='img'>
<img src='/images/lovelace/lovelace_entity_button_card.png' alt='Screenshot of three Button Cards'>
Screenshot of three Button Cards.
</p>

### Card Settings

{% configuration_basic %}
Entity:

>   description: The entity ID the card interacts with, for example `light.living_room`.

Reviewer 2: ```suggestion
  description: The entity ID the card interacts with, for example, `light.living_room`.
```

Name:
  description: The button name that is displayed on the card. If this field is left blank and the card interacts with an entity, the button name defaults to the entity name. Otherwise, no name is displayed.
Icon:
  description: The icon that is displayed on the card. If this field is left blank and the card interacts with an entity, the icon defaults to the entity domain icon. Otherwise, no icon is displayed.
Show Name:
  description: A toggle to show or hide the button name.
Show Icon:
  description: A toggle to show or hide the icon.
Icon Height:
  description: The height of the icon, in pixels.
Theme:
  description: The card theme, which may be set to any installed theme. For more information about themes, see the [frontend documentation](https://www.home-assistant.io/integrations/frontend/).
Tap Action:
  description: The action taken on card tap. For more information, see the [action documentation](/lovelace/actions/#tap-action).
Hold Action:
  description: The action taken on card tap and hold. For more information, see the [action documentation](/lovelace/actions/#hold-action).
{% endconfiguration_basic %}

## Code Editor or YAML mode

> Use these configuration variables if you prefer to use the Code Editor to configure your cards, or if you use YAML mode.

Reviewer 2: ```suggestion
Use these configuration variables if you prefer to use the Code Editor to configure your cards or using YAML mode.
```

Reviewer 2: Meh... not sure either... if you.. if you.. felt a bit weird...

Author: What about, “Use these YAML configuration variables if you are setting up your cards in the Code Editor or in YAML mode.”

Screenshot of the Button card.
</p>

> ## Visual Editor

Reviewer: ```suggestion
## Card Settings
```

Author: Technically, all these variables (including the YAML configuration variables) are card settings - in one case we are just changing them in a visual editor, and in another case we are changing them either in the code editor using YAML or in YAML mode.  Could we maybe have something like this?
![90C31047-D9B3-4CE1-9F81-CB55CAA8A6A9](https://user-images.githubusercontent.com/64062738/88123637-69d76e00-cc1f-11ea-98fc-ee26a0f39995.jpeg)
![0BA85689-B965-494C-B873-D2ACB8C1556D](https://user-images.githubusercontent.com/64062738/88123646-6e9c2200-cc1f-11ea-835f-36a1d01225f8.jpeg)

Reviewer: I do like the `Code Editor or YAML Mode`

But I still don't think we need Visual Editor. To a user coming in they don't know that they are using a visual editor. They just know they are having trouble with the card. The Card Settings are for both and explain the card as a whole. Therefore the first thing a user sees should be the settings covering the card. Even if they are using YAML mode. They may not even need to scroll down if they just need a quick explanation on the setting as the Card Settings are for both

Author: Yea that makes sense. I’ll leave in the Code Editor heading and remove the Visual Editor bit.

<p class='img'>

> <img src='https://i.imgur.com/HJybKBf.jpg' alt='Screenshot of the button card Visual Editor'>Screenshot of the Button card Visual Editor</p>

Reviewer: Lets remove the editor image. Images get out of date easily

### Configuration

{% configuration_basic %}
Entity:

>   description: Choose a Home Assistant entity from the dropdown menu

Reviewer: Same as the Name below

Something like
```
The entity you would like the button to interact with

Name:

>   description: Enter a name for the button. The default is the entity name.

Reviewer: Instead of an action. Describe the field instead.

Something to the degree of
```
The Name that will show on the button
```

Icon:

>   description: Choose an icon to overwrite the default entity picture or entity icon.

Reviewer: Same here, not an action but describing what the field does/is

Show Name:
  description: Toggle to show or hide the button name.
Show Icon:
  description: Toggle to show or hide the button icon.
Theme:

>   description: Choose a card theme from the dropdown menu.

Reviewer: Same here

Tap Action:

>   description: Choose an action from the dropdown menu. This action is taken on card tap. See [action documentation](/lovelace/actions/#tap-action).

Reviewer: same here

Hold Action:

>   description: Choose an action from the dropdown menu. This action is taken on card tap and hold. See [action documentation](/lovelace/actions/#hold-action).

Reviewer: same

Screenshot of the Button card.
</p>

## Visual Editor

<p class='img'>
<img src='https://i.imgur.com/HJybKBf.jpg' alt='Screenshot of the button card Visual Editor'>

### Fields

- **Entity**. Choose a Home Assistant entity from the dropdown menu.
- **Name**. Enter a name for the button. The default is the entity name.
- **Icon**. Choose an icon to overwrite the default entity picture or entity icon. 
- **Show Name?** Toggle to show or hide the button name. 
- **Show Icon?** Toggle to show or hide the button icon.
- **Icon Height**. Enter a custom height for the icon. The height is automatically set if no height is entered. 
- **Theme**. Choose a card theme from the dropdown menu.
- **Tap Action**. Choose an action from the dropdown menu. This action is taken on card tap. See [action documentation](/lovelace/actions/#tap-action).
- **Hold Action**. Choose an action from the dropdown menu. This action is taken on card tap and hold. See [action documentation](/lovelace/actions/#hold-action).

## YAML

### Required Variables

```yaml
type: button
entity: light.living_room
type: button # string
```

> {% configuration %}

Reviewer 2: We should not remove the configuration blocks. Instead, we should improve on their output.

Author: Ok no problem. 

So just to clarify; are we trying to change the configuration block output on all pages that contain variables, or only for the Cards documentation? 

Definitely happy to keep working on this, however I'm not super familiar with Jekyll/Liquid. Where would I go to be able to change how the configuration block is output?

Reviewer 2: Configuration is generated by this plugin: <https://github.com/home-assistant/home-assistant.io/blob/current/plugins/configuration.rb>

The styling can be found in the `sass` folder.

## Code Editor

```yaml
type: button
entity: light.living_room
```

{% configuration %}
type:
  required: true
  description: button
  type: string
entity:
  required: false

>   description: Selects an entity to be used for the card. Use the Home Assistant entity ID, for example `lights.bedroom`.

Reviewer: ```suggestion
  description: The entity to be used for the card. Use the Home Assistant entity ID, for example `lights.bedroom`.
```

  type: string
name:
  required: false

>   description: Sets a name to overwrite the default entity name.

Reviewer: ```suggestion
  description: The name of the button. If not set, the name will be the name of the entity chosen
```

  type: string
  default: Name of Entity
icon:
  required: false

>   description: Selects an icon to overwrite the default entity picture or entity icon.

Reviewer: ```suggestion
  description: The icon on the button. If not set, the icon will be the picture or icon set on the entity chosen.
```

  type: string
entity:
  required: false

>   description: The entity the card interacts with, for example `light.living_room`.

Reviewer: ```suggestion
  description: The entity ID the card interacts with, for example `light.living_room`.
```

  type: string
name:
  required: false
  description: The button name.
  type: string
  default: Entity name
icon:
  required: false
  description: The icon shown on the card.
  type: string
  default: Entity Domain Icon
show_name:
  required: false

>   description: If true, the button name is shown on the card.

Reviewer: We should flip the logic of the boolean descriptions. since its default true. Let them know what happens if its false

Same for show_icon

  type: boolean
  default: "true"
show_icon:
  required: false
  description: If true, the icon is shown on the card.
  type: boolean
  default: "true"
icon_height:
  required: false
  description: The height of the icon. Any CSS value may be used.
  type: string
  default: auto
state_color:
  required: false

>   description: If true, the icon changes color when the entity is active.

Reviewer: Here as well

  type: boolean
  default: true
tap_action:
  required: false
  description: The action taken on card tap. For more information, see the [action documentation](/lovelace/actions/#tap-action).
  type: map
hold_action:
  required: false

>   description: The action taken on card tap and hold. For more information, see the [action documentation](/lovelace/actions/#hold-action).

Reviewer: Tap and hold?

  type: string
entity:
  required: false

>   description: The entity ID the card interacts with, for example `light.living_room`.

Reviewer 2: ```suggestion
  description: The entity ID the card interacts with, for example, `light.living_room`.
```

  default: Entity name
icon:
  required: false
  description: The icon that is displayed on the card. It defaults to the entity domain icon only if the card interacts with an entity. Otherwise, if not configured, no icon is displayed.
  type: string
  default: Entity Domain Icon
show_name:
  required: false
  description: If false, the button name is not shown on the card.
  type: boolean
  default: "true"
show_icon:
  required: false
  description: If false, the icon is not shown on the card.
  type: boolean
  default: "true"
icon_height:
  required: false
  description: The height of the icon. Any CSS value may be used.
  type: string
  default: auto
state_color:
  required: false
  description: If false, the icon does not change color when the entity is active.
  type: boolean
  default: true
tap_action:
  required: false
  description: The action taken on card tap. For more information, see the [action documentation](/lovelace/actions/#tap-action).
  type: map
hold_action:
  required: false
  description: The action taken on card tap and hold. For more information, see the [action documentation](/lovelace/actions/#hold-action).
  type: map
double_tap_action:
  required: false

>   description: The action taken on card double tap. For more information, see the [action documentation](/lovelace/actions/#double-tap-action).

Reviewer 2: ```suggestion
  description: The action taken on card double-tap. For more information, see the [action documentation](/lovelace/actions/#double-tap-action).
```

<p class='img'>

> <img src='/images/lovelace/lovelace_entity_button_complex_card.png' alt='Screenshot of the button card'>

Reviewer: Align the alt with dscr
